### PR TITLE
feat: searchable selector form component 

### PR DESCRIPTION
**Pull request by jonasbrunvoll**
### Background
In drop-down lists with many options, it is desirable to have a search field to filter the list alternatives.  To achieve this, the component `SearchableSelector` is created.  

#### Illustrations
<details>
<summary>screenshots</summary>

| Light    | Dark |
| --------| ------- |
| Skjermbilde 2024-10-18 kl  08 18 36 | Skjermbilde 2024-10-18 kl  08 19 43 |
| Skjermbilde 2024-10-18 kl  08 18 43 | Skjermbilde 2024-10-18 kl  08 19 50 |
| Skjermbilde 2024-10-18 kl  08 19 04 | Skjermbilde 2024-10-18 kl  08 20 00 |

</details>

### Proposed solution
The `SearchableSelector` enables a user to do the following: 
- [x] Select an option from the as the existing `Select` component.
- [x] Filter the list of options by typing the search word into an input field. 

The `SearchableSelector` also ensures the following:
- [x] Updates the specified `state.context.values`. 
- [x] searchable selector´s that depends on other values to be activated, are disabled whenever the dependent value is set back to `undefined`.

### CODE REVIEW

File: src/page-modules/contact/components/input/description-modal.tsx [hunk 1]
```diff
@@ -43,7 +43,7 @@ const DescriptionModal = ({
       document.removeEventListener('keydown', handleEscapeOrClickOutside);
       document.removeEventListener('mousedown', handleEscapeOrClickOutside);
     };
-  }, []);
+  });
```
Comment (mortennordseth) [inline src/page-modules/contact/components/input/description-modal.tsx:46]: Why is this changed and why remove the empty dependency list?

File: src/page-modules/contact/utils.ts [hunk 2]
```diff
@@ -75,6 +77,20 @@ export const setLineAndResetStops = (context: any, line: Line) => {
...
+};
+
+export const getStopOptions = (
+  stops: Line['quays'],
+): Option<Line['quays'][0]>[] => {
+  return stops.map((stop) => ({
+    id: stop.id,
+    name: stop.name,
+    value: stop,
+  }));
```
Comment (mortennordseth) [inline src/page-modules/contact/utils.ts:95]: These utility functions are very specific to the new search component, so I wouldn't place it in this file but rather have it closer to the component itself

File: src/page-modules/contact/components/input/description-modal.tsx (hunk 1 above)
Comment (jonasbrunvoll) [reply src/page-modules/contact/components/input/description-modal.tsx:46]: I received a warning when the deployment ran `yarn tsc` during deployment. I don't think it makes any difference if I remove  the empty dependency list or not, but I can undo the change to keep order.

File: src/page-modules/contact/utils.ts (hunk 2 above)
Comment (jonasbrunvoll) [reply src/page-modules/contact/utils.ts:95]: Maybe create a folder names `searchable-selector` inside `components` containing a `utils.ts` with the functions?

File: src/page-modules/contact/components/input/description-modal.tsx (hunk 1 above)
Comment (mortennordseth) [reply src/page-modules/contact/components/input/description-modal.tsx:46]: We use empty dependency arrays for useEffects if we want the useEffect to run once (on initial render). Else, you want to add the dependencies to the array so that it runs when they changes (i.e. `handleEscapeOrClickOutside` in this case). I guess the warning you get is from es lint and not typescript, but you can disable it using `// eslint-disable-line react-hooks/exhaustive-deps` if you want to use an empty dependency list

File: src/page-modules/contact/components/input/searchable-select.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+              className={style.searchable_select__listBoxItem}
+            >
+              {({ isSelected, isFocused }) => (
+                <div
+                  className={`${style.searchable_select__item} ${
+                    isFocused ? style.searchable_select__highlight : ''
+                  } ${isSelected ? style.searchable_select__selected : ''}`}
+                >
+                  <span className={style.searchable_select__checkmark}>
+                    {value && isSelected ? '✔' : '\u00A0'}
```
Comment (mortennordseth) [inline src/page-modules/contact/components/input/searchable-select.tsx:133]: Could we use the `Confirm.svg` instead here? https://github.com/AtB-AS/design-system/blob/main/packages/assets/files/common/mono/actions/Confirm.svg

Comment (jonasbrunvoll) [reply src/page-modules/contact/components/input/searchable-select.tsx]: Sure, we can use `Confirm.svg` instead. Will update.

**Review by mortennordseth — APPROVED** [review]
